src/backend/columnar/columnar_tableam.c

```diff
@@ -1071,6 +1092,74 @@ columnar_vacuum_rel(Relation rel, VacuumParams *params,
 	{
 		TruncateColumnar(rel, elevel);
 	}
+
+	BlockNumber new_rel_pages = smgrnblocks(RelationGetSmgr(rel), MAIN_FORKNUM);
+
+	/* get the number of indexes */
+	Relation *indrels = NULL;
+	int nindexes;
+	vac_open_indexes(rel, RowExclusiveLock, &nindexes, &indrels);
+	vac_close_indexes(nindexes, indrels, NoLock);
+
```

Comment:
Should we do someting with indRels ? Since this code currently just opens and closes those indexes.

Reply:
vac_open_indexes seems() to be skipping invalid indexes, so might be nice to add a test where we vacuum a relation that has an invalid index ?

Reply:
I changed to use `RelationGetIndexList()` instead, which is lighter-weight.

It doesn't look like it's skipping indexes where `!indisvalid`... the comment over `vac_open_indexes()` says:

```
 * We consider an index vacuumable if it is marked insertable (indisready).                                                
 * If it isn't, probably a CREATE INDEX CONCURRENTLY command failed early in                                               
 * execution, and what we have is too corrupt to be processable.  We will                                                  
 * vacuum even if the index isn't indisvalid; this is important because in a                                               
 * unique index, uniqueness checks will be performed anyway and had better not                                             
 * hit dangling index pointers.                                                                                            
```

If you mean try to test the case where `!indisready`, that seems like overkill, unless there's a case I'm missing.